File: 9020.cpp
#include <iostream>
#include <cstdlib>
using namespace std;

bool prime[10000];

void isPrime(int n)
{
	prime[1] = true;

	for (int i = 2; i <= n; i++) {
		if (prime[i])
			continue;
		for (int j = i * i; j <= n; j += i) {
			prime[j] = true;
		}
	}
}

void process()
{
	int n;
	cin >> n;

	isPrime(n);

	int ret = 0;
	int min = 10000;
	for (int i = 2; i < n; i++) {
		if (!prime[i] && !prime[n - i]) {
                        // 두 소수의 차이가 작은 경우
                        // 차가 작은 경우(같은 경우X)에만 ret 값을 바꾸기 때문에 ret은 n-ret보다 작음 
			if (min > abs((n - i) - i)) {   
				min = abs((n - i) - i);
				ret = i;
			}
		}
	}

	cout << ret << " " << n - ret << endl;
}

int main()
{
	int t;
	cin >> t;

	for (int i = 0; i < t; i++) {
		process();
	}

	return 0;
}

File: 9020.py
prime = [True for i in range(10000+1)]

def isPrime():
    prime[1] = False
    prime[0] = False
    for i in range(2, 10000+1):
        if not prime[i]:
            continue
        for j in range(i*i, 10000+1, i):
            prime[j] = False

isPrime()
t = int(input())
for i in range(t):
    n = int(input())

    temp = n // 2
    while True:
        if prime[temp] and prime[n-temp]:
            print(str(temp) + " " + str(n-temp))
            break
        else:
            temp -= 1
